src/test/java/teammates/test/cases/webapi/UpdateInstructorPrivilegeActionTest.java

```diff
@@ -0,0 +1,318 @@
+    @Test
+    protected void testExecute_withCourseIdAndSectionNameAndSessionName_shouldUpdatePrivilegesInSessionLevel() {
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.INSTRUCTOR_ID, instructor1OfCourse1.getGoogleId(),
+                Const.ParamsNames.COURSE_ID, instructor1OfCourse1.courseId,
+        };
+
+        InstructorPrivilegeUpdateRequest reqBody = new InstructorPrivilegeUpdateRequest();
+        reqBody.setCanSubmitSessionInSections(true);
+        reqBody.setSectionName("Tutorial1");
+        reqBody.setFeedbackSessionName("Session1");
+
+        UpdateInstructorPrivilegeAction action = getAction(reqBody, submissionParams);
+
+        JsonResult result = getJsonResult(action);
+        assertEquals(HttpStatus.SC_OK, result.getStatusCode());
+
+        InstructorPrivilegeData response = (InstructorPrivilegeData) result.getOutput();
+        assertTrue(response.isCanModifyCourse());
+        assertTrue(response.isCanModifySession());
+        assertTrue(response.isCanModifyStudent());
+        assertTrue(response.isCanModifyInstructor());
+
+        assertTrue(response.isCanViewStudentInSections());
+
+        assertTrue(response.isCanSubmitSessionInSections());
+        assertFalse(response.isCanViewSessionInSections());
+        assertFalse(response.isCanModifySessionCommentsInSections());
+    }
+
+    @Test
+    protected void testExecute_requestPrivilegesInconsistent_shouldBeAutoFixed() {
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.INSTRUCTOR_ID, instructor1OfCourse1.getGoogleId(),
+                Const.ParamsNames.COURSE_ID, instructor1OfCourse1.courseId,
+        };
+
+        InstructorPrivilegeUpdateRequest reqBody = new InstructorPrivilegeUpdateRequest();
+        reqBody.setCanModifySessionCommentsInSections(true);
+        reqBody.setSectionName("TUT1");
+
+        UpdateInstructorPrivilegeAction action = getAction(reqBody, submissionParams);
+
+        JsonResult result = getJsonResult(action);
+        assertEquals(HttpStatus.SC_OK, result.getStatusCode());
+
+        InstructorPrivilegeData response = (InstructorPrivilegeData) result.getOutput();
+        assertTrue(response.isCanModifyCourse());
+        assertTrue(response.isCanModifySession());
+        assertTrue(response.isCanModifyStudent());
+        assertTrue(response.isCanModifyInstructor());
+
+        assertFalse(response.isCanViewStudentInSections());
+
+        assertFalse(response.isCanSubmitSessionInSections());
+        assertTrue(response.isCanViewSessionInSections());
+        assertTrue(response.isCanModifySessionCommentsInSections());
+    }
+
+    @Test
+    protected void testExecute_lastInstructorWithModifyInstructorPrivilege_shouldPreserve() {
```

Comment:
In this case, I prefer an assertion before the test: It is indeed true that this is the last instructor with modify privilege.